### PR TITLE
refactoring and tests

### PR DESCRIPTION
**Pull request by danieldietzler**
This has gotten bigger than expected lmao

### CODE REVIEW

**Review by bo0tzz — APPROVED** [review]
LGTM, but of the "quick look before bed" type ;)

**Comment by danieldietzler** [comment]
> LGTM, but of the "quick look before bed" type ;)

You can look at it tomorrow again :D

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -28,3 +28,7 @@ jobs:
       - name: Run formatter
         run: npm run format
         if: ${{ !cancelled() }}
+
+      - name: Run unit tests
+        run: npm run test
```
Comment (jrasm91) [inline .github/workflows/test.yml:33]: npm test

File: package.json [hunk 2]
```diff
@@ -14,19 +14,24 @@
     "format": "prettier --check .",
     "format:fix": "prettier --write .",
     "lint": "eslint . --max-warnings 0",
-    "lint:fix": "npm run lint -- --fix"
+    "lint:fix": "npm run lint -- --fix",
+    "test": "vitest",
+    "test:watch": "vitest --watch",
+    "test:cov": "vitest --coverage"
```
Comment (jrasm91) [inline package.json:20]: Does coverage work?

File: package.json [hunk 3]
```diff
@@ -14,19 +14,24 @@
...
+    "@octokit/types": "^13.4.1",
     "@types/lodash": "^4.14.200",
     "@types/luxon": "^3.3.3",
     "cron": "^3.0.0",
     "discord.js": "^14.14.0",
     "discordx": "^11.7.6",
     "lodash": "^4.17.21",
-    "luxon": "^3.4.3"
+    "luxon": "^3.4.3",
+    "vitest": "^1.5.1"
```
Comment (jrasm91) [inline package.json:34]: These should be dev dependencies if they aren't needed at runtime

File: src/service.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        value: String(item.number),
+      };
+    });
+  } catch (error) {
+    console.log('Could not fetch search results from GitHub');
+    return [];
+  }
+}
+
+const _star_history: Record<string, number | undefined> = {};
```
Comment (jrasm91) [inline src/service.ts:30]: Maybe in the future you could move this to the repo so it could easily be replaced with a database later

**Review by jrasm91 — APPROVED** [review]
Nice!

File: src/service.ts (hunk 4 above)
Comment (danieldietzler) [reply src/service.ts:30]: Yeah I wanted to do that real quick but then I've seen it's a little more involved so I'll postpone it ;)

File: src/repositories/bot.repository.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import { Octokit } from '@octokit/rest';
+import { IMMICH_REPOSITORY_BASE_OPTIONS, Constants } from '../constants.js';
+import { RequestError } from '@octokit/request-error';
+
+const octokit = new Octokit();
+export class BotRepository {
```
Comment (bo0tzz) [inline src/repositories/bot.repository.ts:6]: Shouldn't this technically be called `GithubRepository`? ;)

Comment (danieldietzler) [reply src/repositories/bot.repository.ts]: Done!

**Pull request merged**